### PR TITLE
feat: add a global setting 'interactive' for all commands

### PR DESCRIPTION
**Pull request by Alive-Fish**
E2E TEST: https://github.com/OfficeDev/TeamsFx/actions/runs/1610195003

CLI changes:
1. Add `teamsfx config set interactive true/false` to set it globally.
2. Add the `--interactive` argument for all commands which will overwrite the global `interactive` config.
3. When users open the `interactive` flag, CLI will ignore all the inputs that may be used by `fx-core` and output a log for user.
4. When users close the `interactive` flag, CLI will not let user answer any question when executing the command.
5. The `CI_ENABLED` flag is only used internally just for E2E tests. If it opens, it will set `interactive` to `false`.
image
image
image
image

### CODE REVIEW

**Review by chagong — APPROVED** [review]

**Review by MuyangAmigo — APPROVED** [review]
LGTM

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export async function start() {
       boolean: true,
       default: false,
     })
+    .options("interactive", {
+      description: "Run the command interactively.",
```
Comment (IvanJobs) [inline packages/cli/src/index.ts:46]: Nit: An switch to toggle if the command is running interactively or not?

File: packages/cli/src/yargsCommand.ts [hunk 2]
```diff
@@ -34,12 +35,26 @@ export abstract class YargsCommand {
...
+  public params: { [_: string]: Options } = {};
+
   /**
    * builds the command using supplied yargs handle.
    * @param yargs the yargs handle
    */
   abstract builder(yargs: Argv): Argv<any>;
 
+  /**
+   * before running command, some command may modify the arguments that users input.
```
Comment (IvanJobs) [inline packages/cli/src/yargsCommand.ts:50]: Nit: get the context.

**Review by IvanJobs — APPROVED** [review]
lgtm for the description.

**Pull request merged**